lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -1024,9 +1026,35 @@ namespace Js
         return RecyclerNewFinalized(recycler, ProjectionArrayBuffer, buffer, length, type);
     }
 
-    void ProjectionArrayBuffer::Dispose(bool isShutdown)
+    void ProjectionArrayBuffer::Finalize(bool isShutdown)
     {
+        if (isExternal)
```

Comment:
Why don't we want to do CoTaskMemFree if it is external?

Reply:
Before your change, we would have CoTaskMemFree that case too.

---
In reply to: [186237676](https://github.com/Microsoft/ChakraCore/pull/5099#discussion_r186237676) [](ancestors = 186237676)

Reply:
Right, I was not sure about the ownership here. It did look as something that might be owned externally.
As we discussed we will be treating external buffer as completely owned. - will always free in finalize and charge/uncharge to the quota regardless who allocated.